### ISSUE TITLE
[Problem] Techdraw dimension lines not being recalculated when changing scale

### ISSUE THREAD

**Issue by kesslwovv** [FreeCAD/FreeCAD#8919]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

0.20 (Release)

### Full version info

```shell
[code]
OS: macOS 10.16
Word size of FreeCAD: 64-bit
Version: 0.20.2.29603 (Git)
Build type: Release
Branch: (HEAD detached at 0.20.2)
Hash: 930dd9a76203a3260b1e6256c70c1c3cad8c5cb8
Python 3.10.8, Qt 5.15.4, Coin 4.0.0, Vtk 9.1.0, OCC 7.6.3
Locale: C/Default (C)
Installed mods: 
  * sheetmetal 0.2.61
  * Help 1.0.3
[/code]
```

### Subproject(s) affected?

Techdraw

### Problem description

The dimension lines are not updated properly when changing the scale (in this case 1:5 to 1:4). 

The distances shown in the text are at first the original distance Texts (not updated), when selected and unselected they show their proper distances (they measure the distance between the arrows) but the origin of the line is not anchored to the original vertex.

I found that the following procedure fixes this:
double-clicked the dimension line to open the edit menu
press 'cancel' to exit the menu again

now it is showing the lines properly.

### Anything else?

Wrong (after selecting and unselecting the lines):
![Bildschirm­foto 2023-03-16 um 22 08 14](https://user-images.githubusercontent.com/118482384/225753825-eb4b655e-da6a-423c-87a9-6caee3ff93df.png)

Fixed with above Method:
![Bildschirm­foto 2023-03-16 um 22 18 25](https://user-images.githubusercontent.com/118482384/225755145-63c96818-9f35-4b6a-a596-cba24d0d7033.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by WandererFan** [comment]
I'm having trouble duplicating this behaviour.  What scale are you changing?  Please attach your file.
![dimNotScaling8919](https://user-images.githubusercontent.com/3587161/226191808-49c00a1e-4ca2-46a2-ad93-726b19c63268.png)

**Comment by kesslwovv** [comment]
I am talking about the user-defined scale option in the projection group.

I made a minimal example (basically a rectangle sketch with a pad):

This is as expected (scale 1:1)
![Bildschirm­foto 2023-03-19 um 18 56 17](https://user-images.githubusercontent.com/118482384/226197659-e6556e37-456a-4f97-bc49-d5161c303c0c.png)

This is when changing the scale to 1:2
![Bildschirm­foto 2023-03-19 um 18 56 46](https://user-images.githubusercontent.com/118482384/226197664-80d280bc-83ed-4581-82d8-c73787b8c125.png)

I re-opened the saved file (looks normal) and then changed the scale back to 1:1
![Bildschirm­foto 2023-03-19 um 18 59 42](https://user-images.githubusercontent.com/118482384/226197666-f077b079-d089-4050-97d6-c1015bfc8746.png)

This is the menu option where I change the scale (in ProjGroup). Note that This is using user-defined scale mode.
![Bildschirm­foto 2023-03-19 um 18 57 17](https://user-images.githubusercontent.com/118482384/226197665-d92b5aae-0e2c-4cc6-9bc9-188d8d515996.png)

this is the test file:
[test.FCStd.zip](https://github.com/FreeCAD/FreeCAD/files/11011910/test.FCStd.zip)

**Comment by WandererFan** [comment]
OK, I've got it now.  There is a missing recompute somewhere.   

https://wiki.freecad.org/TechDraw_RedrawPage will fix the display.  

Seems to be fixed in v0.21, but I'll take a look to see if it is an easy backport to v0.20.

**Comment by kesslwovv** [comment]
thanks. I didn't know this option existed (I was trying the default refresh option but this didn't work)
